src/apps/explorer/components/OrderWidget/view.tsx

```diff
@@ -14,27 +12,26 @@ const Wrapper = styled.div`
 `
 
 export type Props = {
-  order: RawOrder | null
+  order: Order | null
   isLoading: boolean
-  error?: string
-  buyToken?: TokenErc20
-  sellToken?: TokenErc20
+  errors: Record<string, string>
 }
 
 export const OrderWidgetView: React.FC<Props> = (props) => {
-  const { order, isLoading, error, buyToken, sellToken } = props
+  const { order, isLoading, errors } = props
 
   return (
     <Wrapper>
       <h2>Order details</h2>
       {/* TODO: create common loading indicator */}
-      {isLoading && <FontAwesomeIcon icon={faSpinner} spin size="5x" />}
-      {order && !isLoading && buyToken && sellToken && (
-        <OrderDetails order={order} buyToken={buyToken} sellToken={sellToken} />
-      )}
+      {order?.buyToken && order?.sellToken && <OrderDetails order={order} />}
       {!order && !isLoading && <p>Order not found</p>}
+      {!isLoading && order && (!order?.buyToken || !order?.sellToken) && <p>Not able to load tokens</p>}
```

Comment:
don't think you need `order` here if you use the `nullish coalesence` of `!order?.buToken` and sell

Reply:
I actually need it. Only want to show it when one of the tokens failed loading.
Otherwise when the order is not found both lines 28 and 29 will be rendered.